include/libtorrent/announce_entry.hpp

```diff
@@ -62,13 +91,7 @@ namespace libtorrent
 		// trackerid is sent).
 		std::string trackerid;
 
-		// if this tracker has returned an error or warning message
-		// that message is stored here
-		std::string message;
-
-		// if this tracker failed the last time it was contacted
-		// this error code specifies what error occurred
-		error_code last_error;
+		std::vector<announce_endpoint> endpoints;
```

Comment:
A problem with treating every (tracker, listen socket) pair as a separate tracker is it interacts poorly with the `announce_to_all_tiers` setting. If it is set to true we will only announce to a single tracker with a single listen socket per tier, which is not what we want. Perhaps the setting should be deprecated?